rqt_gps_rtk_plugin/include/rqt_gps_rtk_plugin/GpsRtkPluginThreads.hpp

```diff
@@ -0,0 +1,185 @@
+//
+// Created by mpantic on 07.03.19.
+//
+
+#ifndef RQT_GPS_RTK_PLUGIN_GPSRTKPLUGINTHREADS_H
+#define RQT_GPS_RTK_PLUGIN_GPSRTKPLUGINTHREADS_H
+#include <QThread>
+#include <rqt_gps_rtk_plugin/SBPStreamDecoder.hpp>
+#include <atomic>
+
+//serial port
+#include <unistd.h> // UNIX standard function definitions
+#include <fcntl.h> // File control definitions
+#include <termios.h> // POSIX terminal control definitions
+#include <sstream>
+
+// udp
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+
+// Thread to read corrections from UDP
+
+class UDPThread : public QThread {
+ Q_OBJECT
+
+ public:
+  UDPThread() : QThread(),
+                fd_socket_(0),
+                stop_thread_(true),
+                udp_port_(26078) {}
```

Comment:
How does it know which interface or subnet to sniff?

Reply:
It listens on all interface on 26078. 
Advantages:
Can't put the wrong interface :)
Disadvantages:
If there is something else on 26078 on another interface, it might cause a bit higher cpu usage (for checking and kicking out non-sbp messages). But very low probability that this happens.